connect/types.go

```diff
@@ -0,0 +1,175 @@
+package connect
+
+
+type GandalfErrorCode int
+
+// GandalfError is a custom error type for validation errors
+type GandalfError struct {
+	Message string
+	Code    GandalfErrorCode
+}
+
+type Application struct {
+	// The human-readable name of the application.
+	AppName string `json:"appName"`
+	// A public key associated with the application, used for cryptographic operations such as
+	// verifying the identity of the application.
+	PublicKey string `json:"publicKey"`
+	// The URL pointing to the icon graphic for the application. This URL should link to an image
+	// that visually represents the application, aiding in its identification and branding.
+	IconURL string `json:"iconURL"`
+	// A unique identifier assigned to the application upon registration.
+	GandalfID int64 `json:"gandalfID"`
+	// The address of the user who registered the application.
+	AppRegistrar string `json:"appRegistrar"`
+}
+// type Application map[string]interface{}
+
+type SupportedService struct {
+	Name              string `json:"name"`
+	Description       string `json:"description"`
+	IsDeprecated      bool   `json:"isDeprecated"`
+	DeprecationReason string `json:"deprecationReason"`
+}
+
+type SupportedServices []Value
+
+type Service struct {
+	Name string
+	Status bool
+}
+
+type Services []Service
+
+// Type represents a GraphQL type with various properties like kind, name, description, etc.
```

Comment:
This types and introspectQuery already exist inside eyeofsauron, you either leave it inside eyeofsauron , or move it outside of it as a module so that both eyeofsauron and connect can reuse this

Reply:
@Bashorun97 this has not been resolved yet. Types and IntrospectResult are still duplicated in both sauron and connect